app/web/src/components/Dialog.tsx

```diff
@@ -69,7 +77,25 @@ export function DialogContentText(props: DialogContentTextProps) {
   );
 }
 
+interface DialogTitleProps extends MuiDialogTitleProps {
+  onClose?: () => void;
+}
+
 export function DialogTitle(props: Omit<DialogTitleProps, "className">) {
```

Comment:
Two things:
- (minor) we usually prefer to destructure in the function parameter rather than another line of code (might want to Prettier format after if applying suggestion directly...)
- It's clearer to reference the type of `props` as `DialogTitleProps` rather than utility type - the required work should be done in the `DialogTitleProps` type itself

```suggestion
export function DialogTitle({ children, onClose, ...dialogTitleProps }: DialogTitleProps) {
```